src/test/java/bio/overture/ego/controller/PolicyControllerTest.java

```diff
@@ -0,0 +1,295 @@
+
+  /** Constants */
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  /** State */
+  @LocalServerPort private int port;
+
+  private TestRestTemplate restTemplate = new TestRestTemplate();
+  private HttpHeaders headers = new HttpHeaders();
+  private static boolean hasRunEntitySetup = false;
+
+  /** Dependencies */
+  @Autowired private EntityGenerator entityGenerator;
+
+  @Autowired PolicyService policyService;
+
+  @Before
+  public void setup() {
+    // Initial setup of entities (run once
+    if (!hasRunEntitySetup) {
+      entityGenerator.setupTestUsers();
+      entityGenerator.setupTestGroups();
+      entityGenerator.setupTestPolicies();
+      hasRunEntitySetup = true;
+    }
+
+    headers.add("Authorization", "Bearer TestToken");
+    headers.setContentType(MediaType.APPLICATION_JSON);
+  }
+
+  @Test
+  @SneakyThrows
+  public void addPolicy() {
+    val policy = Policy.builder().name("AddPolicy").build();
+
+    val entity = new HttpEntity<Policy>(policy, headers);
+
+    val response =
+        restTemplate.exchange(
+            createURLWithPort("/policies"), HttpMethod.POST, entity, String.class);
+
+    val responseStatus = response.getStatusCode();
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    val responseJson = MAPPER.readTree(response.getBody());
+
+    log.info(response.getBody());
+
+    assertThat(responseJson.get("name").asText()).isEqualTo("AddPolicy");
+  }
+
+  @Test
+  @SneakyThrows
+  public void addUniquePolicy() {
+    val policy1 = Policy.builder().name("PolicyUnique").build();
+    val policy2 = Policy.builder().name("PolicyUnique").build();
+
+    val entity1 = new HttpEntity<Policy>(policy1, headers);
+    val response1 =
+        restTemplate.exchange(
+            createURLWithPort("/policies"), HttpMethod.POST, entity1, String.class);
+
+    val responseStatus1 = response1.getStatusCode();
+    assertThat(responseStatus1).isEqualTo(HttpStatus.OK);
+
+    val entity2 = new HttpEntity<Policy>(policy2, headers);
+    val response2 =
+        restTemplate.exchange(
+            createURLWithPort("/policies"), HttpMethod.POST, entity2, String.class);
+    val responseStatus2 = response2.getStatusCode();
+    assertThat(responseStatus2).isEqualTo(HttpStatus.CONFLICT);
+  }
+
+  @Test
+  @SneakyThrows
+  public void getPolicy() {
+    val policyId = policyService.getByName("Study001").getId();
+    val entity = new HttpEntity<String>(null, headers);
+    val response =
+        restTemplate.exchange(
+            createURLWithPort(String.format("/policies/%s", policyId)),
+            HttpMethod.GET,
+            entity,
+            String.class);
+
+    val responseStatus = response.getStatusCode();
+    val responseJson = MAPPER.readTree(response.getBody());
+
+    assertThat(responseStatus).isEqualTo(HttpStatus.OK);
+    assertThat(responseJson.get("name").asText()).isEqualTo("Study001");
+  }
+
+  @Test
+  @SneakyThrows
+  public void addGroupPermission() {
```

Comment:
rename test to `associatePermissionsWithGroup_ExistingEntitiesButNonExistingRelationship_Success`